firmware/shared/controls/DI_driverTorqueRequest.h

```diff
@@ -0,0 +1,26 @@
+#pragma once
+#include <cmath>
+#include <iostream>
+
+namespace ctrl {  
+
+// Templating means this function will work on any types that support addition.
+template <typename T, typename B>
+T DI_driverTorqueRequest(T READY_TO_DRIVE, T DI_Error, B AccelPedPos) {
+    T u1 = 0.0;
+    if((!READY_TO_DRIVE) == 1 || DI_Error == 1){ 

```

Comment:
since these two variables will be made to booleans, you should replace `== 1` with `== true`
However, `X ==true` is equivalent to just `X`, so this line can be simplified even more